### PR TITLE
DEV-1748: Link upsells to subscriptions page when feature flagged

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

User-facing:
When a user has the self-upgrade feature flag, the Mailchimp modal, the Max Pages created modal, and the Customize page directs them to the Subscriptions page.

Behind the scenes:
Remove `useFeatureFlags` in favor of the `useUser`/`flagIsActiveForUser` combo. This is different from the approach I was originally thinking (moving `flagIsActiveForUser` into `useFeatureFlags`), but there are places where we want to work with user feature flags without having to keep calling `useUser` in every component. To me it is cleaner to pass a user as a prop and let components work with that.

#### Why are we doing this? How does it help us?

User-facing:
Improves the self-upgrade process.

Behind the scenes:
Simplifies the approach to feature flags. Prevents some test weirdness where you could potentially mock `useUser` and `useFeatureFlags` to return different results, which would never happen in real life.

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

Not yet--this is all behind a feature flag.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-1748

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

File: spa/src/components/common/Modal/MaxPagesReachedModal/MaxPagesReachedModal.tsx [hunk 1]
```diff
@@ -33,7 +38,32 @@ export interface MaxPagesReachedModalProps extends InferProps<typeof MaxPagesRea
...
+  // The destination of the Upgrade button may be either internal or external.
+
+  const LooseButton = Button as any;
+  let upgradeButton = (
+    <LinkButton color="primaryDark" href={PRICING_URL} target="_blank">
+      Upgrade
+    </LinkButton>
+  );
+
+  if (recommendedPlan === 'CORE') {
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/common/Modal/MaxPagesReachedModal/MaxPagesReachedModal.tsx]: Yup--either missed this originally or did an incorrect merge with main. Fixed in fb51e3bad2396af062f5a5622c5a5b887c702720

File: spa/src/components/common/Modal/MaxPagesReachedModal/MaxPagesReachedModal.tsx [hunk 2]
```diff
@@ -33,7 +38,32 @@ export interface MaxPagesReachedModalProps extends InferProps<typeof MaxPagesRea
 }
 
 export function MaxPagesReachedModal({ currentPlan, onClose, open, recommendedPlan }: MaxPagesReachedModalProps) {
-  const upgradeUrl = recommendedPlan === PLAN_NAMES.CORE ? CORE_UPGRADE_URL : PRICING_URL;
+  const { user } = useUser();
+
+  // The destination of the Upgrade button may be either internal or external.
+
+  const LooseButton = Button as any;
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/common/Modal/MaxPagesReachedModal/MaxPagesReachedModal.tsx]: :+1: 2da64bcceb8468a306af5fcbfa8ad4324961eab2

File: spa/src/components/common/IntegrationCard/MailchimpIntegrationCard/MailchimpModal/MailchimpModal.tsx [hunk 3]
```diff
@@ -162,7 +173,11 @@ const MailchimpModal = ({
         </CancelButton>
         {
           {
-            [DISPLAY_STATE.FREE]: (
+            [DISPLAY_STATE.FREE]: flagIsActiveForUser(SELF_UPGRADE_ACCESS_FLAG_NAME, user) ? (
+              <RouterLinkButton {...(actionButtonProps as RouterLinkButtonProps)} to={SETTINGS.SUBSCRIPTION}>
```
Comment (GuiMend) [inline spa/src/components/common/IntegrationCard/MailchimpIntegrationCard/MailchimpModal/MailchimpModal.tsx:177]: If we remove the `Partial<ButtonProps>` from line 93 we perhaps don't need this cast `as RouterLinkButtonProps` here.

`const actionButtonProps: Partial<ButtonProps> = {`

Comment (nrh-cklimas) [reply spa/src/components/common/IntegrationCard/MailchimpIntegrationCard/MailchimpModal/MailchimpModal.tsx]: I tried that but it does seem to need a type declaration or TS won't like spreading the props. However, I found one that doesn't require a cast--see ab61df4feebffe9934c4d71ba0c24336157b225a.

**Review by GuiMend — APPROVED** [review]

**Pull request merged**